File: LatEntry3DGJ.js
var LatEntry3DGJ = function (junctionTemplate) {
    this.junctionTemplate = junctionTemplate;

    // DP1 INCOMINGS
    this.dp1Incoming = null;
    this.dp1IncomingExtender = null;
    this.dp1IncomingExtenderCount = null;

    // DP1 OUTGOINGS
    this.dp1Fork = null;

    this.dp1Outgoing1 = null;
    this.dp1Outgoing1Extender1 = null;
    this.dp1Outgoing1Extender2 = null;
    this.dp1Outgoing1Extender = null;
    this.dp1Outgoing1ExtenderCount = null;

    this.dp1Outgoing2 = null;
    this.dp1Outgoing2Extender1 = null;
    this.dp1Outgoing2Extender2 = null;
    this.dp1Outgoing2Extender = null;
    this.dp1Outgoing2ExtenderCount = null;

    // DP 1 Merge
    this.dp1Merge = null;

    this.dp1MergeIncoming = null;
    this.dp1MergeIncomingExtender1 = null;
    this.dp1MergeIncomingExtender2 = null;
    this.dp1MergeIncomingExtender = null;
    this.dp1MergeIncomingExtenderCount = null;

    this.dp1MergeOutgoing = null;
    this.dp1MergeOutgoingExtender1 = null;
    this.dp1MergeOutgoingExtender2 = null;
    this.dp1MergeOutgoingExtender = null;
    this.dp1MergeOutgoingExtenderCount = null;

    // DP2 FORK
    this.dp2Incoming = null;
    this.dp2Fork = null;

    // DP2 Outgoing1
    this.dp2Outgoing1 = null;
    this.dp2Outgoing1Extender1 = null;
    this.dp2Outgoing1Extender2 = null;
    this.dp2Outgoing1Extender = null;
    this.dp2Outgoing1ExtenderCount = null;

    // DP2 Outgoing2
    this.dp2Outgoing2 = null;
    this.dp2Outgoing2Extender1 = null;
    this.dp2Outgoing2Extender2 = null;
    this.dp2Outgoing2ExtenderCount = null;
    this.dp2Outgoing2Extender = null;


}

function parseEntries(data) {
    var lines = data.split('\n');
    var entries = [];
    for (var i = 1; i < lines.length - 1; i++) {
        var line = lines[i].split(';');

        // displayContents(entries);
        //remove possible "",
        //i.e. if line[j] == ""..."" then line = "..."
        for(var j = 0;j<line.length;j++){
            if(line[j].charAt(0)=='\"'){
                line[j]=line[j].substr(1,line[j].length-2);
            }

        }

        var entry = new LatEntry3DGJ();
        entry.junctionTemplate = line[6];
        entry.dp1Incoming = line[17];
        entry.dp1IncomingExtender = line[18];
        entry.dp1IncomingExtenderCount = line[19];
        entry.dp1Fork = line[20];
        entry.dp1Outgoing1 = line[21];
        entry.dp1Outgoing1Extender1 = line[22];
        entry.dp1Outgoing1Extender2 = line[23];
        entry.dp1Outgoing1Extender = line[24];
        entry.dp1Outgoing1ExtenderCount = line[25];
        entry.dp1Outgoing2 = line[26];
        entry.dp1Outgoing2Extender1 = line[27];
        entry.dp1Outgoing2Extender2 = line[28];
        entry.dp1Outgoing2Extender = line[29];
        entry.dp1Outgoing2ExtenderCount = line[30];
        entry.dp1Merge = line[31];
        entry.dp1MergeIncoming = line[32];
        entry.dp1MergeIncomingExtender1 = line[33];
        entry.dp1MergeIncomingExtender2 = line[34];
        entry.dp1MergeIncomingExtender = line[35];
        entry.dp1MergeIncomingExtenderCount = line[36];
        entry.dp1MergeOutgoing = line[37];
        entry.dp1MergeOutgoingExtender1 = line[38];
        entry.dp1MergeOutgoingExtender2 = line[39];
        entry.dp1MergeOutgoingExtender = line[40];
        entry.dp1MergeOutgoingExtenderCount = line[41];
        entry.dp2Incoming = line[42];
        entry.dp2Fork = line[43];
        entry.dp2Outgoing1 = line[44];
        entry.dp2Outgoing1Extender1 = line[45];
        entry.dp2Outgoing1Extender2 = line[46];
        entry.dp2Outgoing1Extender = line[47];
        entry.dp2Outgoing1ExtenderCount = line[48];
        entry.dp2Outgoing2 = line[49];
        entry.dp2Outgoing2Extender1 = line[50];
        entry.dp2Outgoing2Extender2 = line[51];
        entry.dp2Outgoing2Extender = line[52];
        entry.dp2Outgoing2ExtenderCount = line[53];
        entries.push(entry);

    }
    return entries;
}

File: LatEntry3DGJ.java
package com.navteq.vps.gjv.latcreation3dgj.data;

import com.navteq.vps.gjv.latcreation3dgj.parsingJunctionTemplate.JunctionComponentParser;

public class LatEntry3DGJ
{
    private String junctionTemplate;

    // DP1 INCOMINGS
    private String dp1Incoming;
    private String dp1IncomingExtender;
    private String dp1IncomingExtenderCount;

    // DP1 OUTGOINGS
    private String dp1Fork;

    private String dp1Outgoing1;
    private String dp1Outgoing1Extender1;
    private String dp1Outgoing1Extender2;
    private String dp1Outgoing1Extender;
    private String dp1Outgoing1ExtenderCount;

    private String dp1Outgoing2;
    private String dp1Outgoing2Extender1;
    private String dp1Outgoing2Extender2;
    private String dp1Outgoing2Extender;
    private String dp1Outgoing2ExtenderCount;

    // DP 1 Merge
    private String dp1Merge;

    private String dp1MergeIncoming;
    private String dp1MergeIncomingExtender1;
    private String dp1MergeIncomingExtender2;
    private String dp1MergeIncomingExtender;
    private String dp1MergeIncomingExtenderCount;

    private String dp1MergeOutgoing;
    private String dp1MergeOutgoingExtender1;
    private String dp1MergeOutgoingExtender2;
    private String dp1MergeOutgoingExtender;
    private String dp1MergeOutgoingExtenderCount;

    // DP2 FORK
    private String dp2Incoming;
    private String dp2Fork;

    // DP2 Outgoing1
    private String dp2Outgoing1;
    private String dp2Outgoing1Extender1;
    private String dp2Outgoing1Extender2;
    private String dp2Outgoing1Extender;
    private String dp2Outgoing1ExtenderCount;

    // DP2 Outgoing2
    private String dp2Outgoing2;
    private String dp2Outgoing2Extender1;
    private String dp2Outgoing2Extender2;
    private String dp2Outgoing2ExtenderCount;
    private String dp2Outgoing2Extender;

    public LatEntry3DGJ(String junctionTemplate)
    {
        JunctionComponentParser jcp = new JunctionComponentParser();
        jcp.parseString(this, junctionTemplate);
    }

    public String getDp1IncomingExtender()
    {
        return dp1IncomingExtender;
    }

    public void setDp1IncomingExtender(String dp1IncomingExtender)
    {
        this.dp1IncomingExtender = dp1IncomingExtender;
    }

    public String getDp1Fork()
    {
        return dp1Fork;
    }

    public void setDp1Fork(String dp1Fork)
    {
        this.dp1Fork = dp1Fork;
    }

    public String getDp1Incoming()
    {
        return dp1Incoming;
    }

    public void setDp1Incoming(String dp1Incoming)
    {
        this.dp1Incoming = dp1Incoming;
    }

    public String getDp1IncomingExtenderCount()
    {
        return dp1IncomingExtenderCount;
    }

    public void setDp1IncomingExtenderCount(String dp1IncomingExtenderCount)
    {
        this.dp1IncomingExtenderCount = dp1IncomingExtenderCount;
    }

    public String getDp1Merge()
    {
        return dp1Merge;
    }

    public void setDp1Merge(String dp1Merge)
    {
        this.dp1Merge = dp1Merge;
    }

    public String getDp1MergeIncoming()
    {
        return dp1MergeIncoming;
    }

    public void setDp1MergeIncoming(String dp1MergeIncoming)
    {
        this.dp1MergeIncoming = dp1MergeIncoming;
    }

    public String getDp1MergeIncomingExtender1()
    {
        return dp1MergeIncomingExtender1;
    }

    public void setDp1MergeIncomingExtender1(String dp1MergeIncomingExtender1)
    {
        this.dp1MergeIncomingExtender1 = dp1MergeIncomingExtender1;
    }

    public String getDp1MergeIncomingExtender2()
    {
        return dp1MergeIncomingExtender2;
    }

    public void setDp1MergeIncomingExtender2(String dp1MergeIncomingExtender2)
    {
        this.dp1MergeIncomingExtender2 = dp1MergeIncomingExtender2;
    }

    public String getDp1MergeIncomingExtender()
    {
        return dp1MergeIncomingExtender;
    }

    public void setDp1MergeIncomingExtender(String dp1MergeIncomingExtender)
    {
        this.dp1MergeIncomingExtender = dp1MergeIncomingExtender;
    }

    public String getDp1MergeIncomingExtenderCount()
    {
        return dp1MergeIncomingExtenderCount;
    }

    public void setDp1MergeIncomingExtenderCount(String dp1MergeIncomingExtenderCount)
    {
        this.dp1MergeIncomingExtenderCount = dp1MergeIncomingExtenderCount;
    }

    public String getDp1MergeOutgoing()
    {
        return dp1MergeOutgoing;
    }

    public void setDp1MergeOutgoing(String dp1MergeOutgoing)
    {
        this.dp1MergeOutgoing = dp1MergeOutgoing;
    }

    public String getDp1MergeOutgoingExtender1()
    {
        return dp1MergeOutgoingExtender1;
    }

    public void setDp1MergeOutgoingExtender1(String dp1MergeOutgoingExtender1)
    {
        this.dp1MergeOutgoingExtender1 = dp1MergeOutgoingExtender1;
    }

    public String getDp1MergeOutgoingExtender2()
    {
        return dp1MergeOutgoingExtender2;
    }

    public void setDp1MergeOutgoingExtender2(String dp1MergeOutgoingExtender2)
    {
        this.dp1MergeOutgoingExtender2 = dp1MergeOutgoingExtender2;
    }

    public String getDp1MergeOutgoingExtender()
    {
        return dp1MergeOutgoingExtender;
    }

    public void setDp1MergeOutgoingExtender(String dp1MergeOutgoingExtender)
    {
        this.dp1MergeOutgoingExtender = dp1MergeOutgoingExtender;
    }

    public String getDp1MergeOutgoingExtenderCount()
    {
        return dp1MergeOutgoingExtenderCount;
    }

    public void setDp1MergeOutgoingExtenderCount(String dp1MergeOutgoingExtenderCount)
    {
        this.dp1MergeOutgoingExtenderCount = dp1MergeOutgoingExtenderCount;
    }

    public String getDp1Outgoing1()
    {
        return dp1Outgoing1;
    }

    public void setDp1Outgoing1(String dp1Outgoing1)
    {
        this.dp1Outgoing1 = dp1Outgoing1;
    }

    public String getDp1Outgoing1Extender1()
    {
        return dp1Outgoing1Extender1;
    }

    public void setDp1Outgoing1Extender1(String dp1Outgoing1Extender1)
    {
        this.dp1Outgoing1Extender1 = dp1Outgoing1Extender1;
    }

    public String getDp1Outgoing1Extender2()
    {
        return dp1Outgoing1Extender2;
    }

    public void setDp1Outgoing1Extender2(String dp1Outgoing1Extender2)
    {
        this.dp1Outgoing1Extender2 = dp1Outgoing1Extender2;
    }

    public String getDp1Outgoing1Extender()
    {
        return dp1Outgoing1Extender;
    }

    public void setDp1Outgoing1Extender(String dp1Outgoing1Extender)
    {
        this.dp1Outgoing1Extender = dp1Outgoing1Extender;
    }

    public String getDp1Outgoing1ExtenderCount()
    {
        return dp1Outgoing1ExtenderCount;
    }

    public void setDp1Outgoing1ExtenderCount(String dp1Outgoing1ExtenderCount)
    {
        this.dp1Outgoing1ExtenderCount = dp1Outgoing1ExtenderCount;
    }

    public String getDp1Outgoing2()
    {
        return dp1Outgoing2;
    }

    public void setDp1Outgoing2(String dp1Outgoing2)
    {
        this.dp1Outgoing2 = dp1Outgoing2;
    }

    public String getDp1Outgoing2Extender1()
    {
        return dp1Outgoing2Extender1;
    }

    public void setDp1Outgoing2Extender1(String dp1Outgoing2Extender1)
    {
        this.dp1Outgoing2Extender1 = dp1Outgoing2Extender1;
    }

    public String getDp1Outgoing2Extender2()
    {
        return dp1Outgoing2Extender2;
    }

    public void setDp1Outgoing2Extender2(String dp1Outgoing2Extender2)
    {
        this.dp1Outgoing2Extender2 = dp1Outgoing2Extender2;
    }

    public String getDp1Outgoing2Extender()
    {
        return dp1Outgoing2Extender;
    }

    public void setDp1Outgoing2Extender(String dp1Outgoing2Extender)
    {
        this.dp1Outgoing2Extender = dp1Outgoing2Extender;
    }

    public String getDp1Outgoing2ExtenderCount()
    {
        return dp1Outgoing2ExtenderCount;
    }

    public void setDp1Outgoing2ExtenderCount(String dp1Outgoing2ExtenderCount)
    {
        this.dp1Outgoing2ExtenderCount = dp1Outgoing2ExtenderCount;
    }

    public String getDp2Fork()
    {
        return dp2Fork;
    }

    public void setDp2Fork(String dp2Fork)
    {
        this.dp2Fork = dp2Fork;
    }

    public String getDp2Incoming()
    {
        return dp2Incoming;
    }

    public void setDp2Incoming(String dp2Incoming)
    {
        this.dp2Incoming = dp2Incoming;
    }

    public String getDp2Outgoing1()
    {
        return dp2Outgoing1;
    }

    public void setDp2Outgoing1(String dp2Outgoing1)
    {
        this.dp2Outgoing1 = dp2Outgoing1;
    }

    public String getDp2Outgoing1Extender1()
    {
        return dp2Outgoing1Extender1;
    }

    public void setDp2Outgoing1Extender1(String dp2Outgoing1Extender1)
    {
        this.dp2Outgoing1Extender1 = dp2Outgoing1Extender1;
    }

    public String getDp2Outgoing1Extender2()
    {
        return dp2Outgoing1Extender2;
    }

    public void setDp2Outgoing1Extender2(String dp2Outgoing1Extender2)
    {
        this.dp2Outgoing1Extender2 = dp2Outgoing1Extender2;
    }

    public String getDp2Outgoing1Extender()
    {
        return dp2Outgoing1Extender;
    }

    public void setDp2Outgoing1Extender(String dp2Outgoing1Extender)
    {
        this.dp2Outgoing1Extender = dp2Outgoing1Extender;
    }

    public String getDp2Outgoing1ExtenderCount()
    {
        return dp2Outgoing1ExtenderCount;
    }

    public void setDp2Outgoing1ExtenderCount(String dp2Outgoing1ExtenderCount)
    {
        this.dp2Outgoing1ExtenderCount = dp2Outgoing1ExtenderCount;
    }

    public String getDp2Outgoing2()
    {
        return dp2Outgoing2;
    }

    public void setDp2Outgoing2(String dp2Outgoing2)
    {
        this.dp2Outgoing2 = dp2Outgoing2;
    }

    public String getDp2Outgoing2Extender1()
    {
        return dp2Outgoing2Extender1;
    }

    public void setDp2Outgoing2Extender1(String dp2Outgoing2Extender1)
    {
        this.dp2Outgoing2Extender1 = dp2Outgoing2Extender1;
    }

    public String getDp2Outgoing2Extender2()
    {
        return dp2Outgoing2Extender2;
    }

    public void setDp2Outgoing2Extender2(String dp2Outgoing2Extender2)
    {
        this.dp2Outgoing2Extender2 = dp2Outgoing2Extender2;
    }

    public String getDp2Outgoing2Extender()
    {
        return dp2Outgoing2Extender;
    }

    public void setDp2Outgoing2Extender(String dp2Outgoing2Extender)
    {
        this.dp2Outgoing2Extender = dp2Outgoing2Extender;
    }

    public String getDp2Outgoing2ExtenderCount()
    {
        return dp2Outgoing2ExtenderCount;
    }

    public void setDp2Outgoing2ExtenderCount(String dp2Outgoing2ExtenderCount)
    {
        this.dp2Outgoing2ExtenderCount = dp2Outgoing2ExtenderCount;
    }

    public String getJunctionTemplate()
    {
        return junctionTemplate;
    }

    public void setJunctionTemplate(String junctionTemplate)
    {
        this.junctionTemplate = junctionTemplate;
    }

}
